src/backend/distributed/utils/node_metadata.c

```diff
@@ -1331,9 +1331,15 @@ DeleteNodeRow(char *nodeName, int32 nodePort)
 	HeapTuple heapTuple = NULL;
 	SysScanDesc heapScan = NULL;
 	ScanKeyData scanKey[2];
-
 	Relation pgDistNode = heap_open(DistNodeRelationId(), RowExclusiveLock);
 
+	/*
+	 * simple_heap_delete() expects that the caller has at least an
+	 * AccessShareLock on replica identity index.
+	 */
+	Relation replicaIndex = index_open(RelationGetReplicaIndex(pgDistNode),
+									   AccessShareLock);
```

Comment:
I talked with Andres about this, and he was ok with this approach.

Normally deletions go through postgres executor which acquires the necessary locks, but it seems that we cannot do that here because the user might not have enough permissions for executing DML on pg_dist_node so going through executor might error out.